sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -1565,11 +1569,9 @@ private boolean isADFSTenant() {
 
     private InputStream getCertificateInputStream() throws IOException {
         if (certificatePath != null) {
-            return new FileInputStream(certificatePath);
-        } else if (certificate != null) {
-            return certificate;
+            return new BufferedInputStream(new FileInputStream(certificatePath));
```

Comment:
Use a `BufferedInputStream` to prevent reading the file byte-by-byte which has a much higher overhead